.github/workflows/track_dependencies.py

```diff
@@ -136,19 +152,20 @@ def version_as_list(semver):
                 )
             ) or
             # This is a new tag in the same minor SEMVER but with a higher
-            # patch version; we want to update it
+            # patch version on a non default branch; we want to update it
             (
                 args.branch not in default_branches and
                 ver != tags[-1] and
                 (
-                    int(current_version[0]) == int(new_version[0]) and
-                    int(current_version[1]) == int(new_version[1]) and
-                    int(current_version[2]) < int(new_version[2])
+                    int(current_version[0]) == int(current_version_latest_patch[0]) and
+                    int(current_version[1]) == int(current_version_latest_patch[1]) and
+                    int(current_version[2]) < int(current_version_latest_patch[2])
                 )
             )
         ):
-            output += f'{repo},{tags[-1]};'
-            repos_file['repositories'][repo]['version'] = tags[-1]
+            current_version_latest_patch_str = 'v' + '.'.join(current_version_latest_patch)
```

Comment:
Should this be `'v' + '',join...`?
```suggestion
            current_version_latest_patch_str = 'v' + ''.join(current_version_latest_patch)
```

Reply:
No, `current_version_latest_patch` is a list of three int elements `[X, Y, Z]` and the string representation of that should be `vX.Y.Z`